### ISSUE TITLE
preparation of device descriptions

### ISSUE THREAD

**Issue by LFSaw** [ModalityTeam/Modality-toolkit#164]
Could someone please enlighten me on this?
Seems to be quite a lot of flattening and in-place operations here...
I'd imagine that this would benefit from refactoring.
```
// in MKtl:prLoadDeviceDescription
```

```supercollider

		deviceDescriptionArray = this.makeFlatDeviceDescription( deviceDescriptionHierarch );

		deviceDescriptionArray.pairsDo{ |key,elem|
// I thought it is already flat? The name of the above method call implies that...
// also, in-place operation, why?
			this.class.flattenDescription( elem )
		};


// and again pairsDo?
		// create specs
		deviceDescriptionArray.pairsDo {|key, elem|
			var foundSpec =  specs[elem[\spec]];
			if (foundSpec.isNil) {
				warn("Mktl - in description %, el %, spec for '%' is missing! please add it via:"
					"\nMktl.addSpec( '%', [min, max, warp, step, default]);\n"
					.format(name, key, elem[\spec], elem[\spec])
				);
			};
			elem[\specName] = elem[\spec];
			elem[\spec] = this.class.specs[elem[\specName]];
		};

		deviceInfo[\infoMessage] !? _.postln;

```

**Comment by sensestage** [comment]
Will be solved with issue #183

**Comment by adcxyz** [comment]
this part of the refactoring is done now.
